src/snowflake/connector/pandas_tools.py

```diff
@@ -178,6 +183,54 @@ def write_pandas(
     else:
         columns = ",".join(list(df.columns))
 
+    if auto_create_table:
+        file_format_name = None
+        while True:
+            try:
+                file_format_name = (
+                    '"'
+                    + "".join(random.choice(string.ascii_lowercase) for _ in range(5))
+                    + '"'
+                )
+                file_format_sql = (
+                    f"CREATE FILE FORMAT {file_format_name} "
+                    f"/* Python:snowflake.connector.pandas_tools.write_pandas() */ "
+                    f"TYPE=PARQUET COMPRESSION={compression_map[compression]}"
+                )
+                logger.debug("creating file format with '{}'".format(file_format_sql))
```

Comment:
Nit: it might be better to use `logger.debug("creating file format with %r", file_format_sql)`, which avoids the string format when logging level is not debug.

Reply:
Maybe, however f-strings are way faster than any other string interpolation, see this post: https://www.scivision.dev/python-f-string-speed/
But interesting enough:

```
import logging
from timeit import timeit
sql_command="PUT file:///tmp/log.txt @;"
timings = []
# Non-Printing
timings.append(timeit('logging.debug(f"logging message: {sql_command}")', globals=locals()))
timings.append(timeit('logging.debug("logging message: %r", sql_command)', globals=locals()))
# Printing
timings.append(timeit('logging.warning(f"logging message: {sql_command}")', globals=locals()))
timings.append(timeit('logging.warning("logging message: %r", sql_command)', globals=locals()))
```

I get: `[0.5704036249953788, 0.567899767003837, 23.281877171990345, 25.245735585995135]`
So actually the price you pay upon logging is so much bigger than the non-logging price between f-strings and % formatting that I have actually decided to pay the price of always using f-strings for readability purposes.

So please for now follow using f-strings only, if you really want to optimize this then let's have that conversation in another ticket